pkg/generated-assets/generated_assets.go

```diff
@@ -0,0 +1,272 @@
+	StorageClassAssetNames     []string `yaml:"storageClassAssetNames"`
+	VolumeSnapshotClassNames   []string `yaml:"volumeSnapshotClassNames"`
+}
+
+// Save saves the assets to the given directory.
+func (a *CSIDriverAssets) Save(path string) error {
+	if path != "" {
+		if err := os.MkdirAll(path, 0755); err != nil {
+			return err
+		}
+	}
+	m := AssetsManifest{
+		ControllerStaticAssetNames: a.GetControllerStaticAssetNames(),
+		GuestStaticAssetNames:      a.GetGuestStaticAssetNames(),
+		StorageClassAssetNames:     a.GetStorageClassAssetNames(),
+		VolumeSnapshotClassNames:   a.GetVolumeSnapshotClassAssetNames(),
+	}
+
+	// Write the list in manifest file in stable order
+	sort.Strings(m.ControllerStaticAssetNames)
+	sort.Strings(m.GuestStaticAssetNames)
+	sort.Strings(m.StorageClassAssetNames)
+	sort.Strings(m.VolumeSnapshotClassNames)
+
+	mYAML, err := yaml.Marshal(m)
+	if err != nil {
+		return fmt.Errorf("failed to marshall manifests: %w", err)
+	}
+	manifestsFile := filepath.Join(path, manifestFileName)
+	if err := os.WriteFile(manifestsFile, mYAML, 0644); err != nil {
+		return fmt.Errorf("failed to write manifests: %w", err)
+	}
+
+	controllerTemplate, err := Sanitize(a.ControllerTemplate)
+	if err != nil {
+		return fmt.Errorf("failed to sanitize controller.yaml: %w", err)
+	}
+	if err := os.WriteFile(filepath.Join(path, ControllerDeploymentAssetName), controllerTemplate, 0644); err != nil {
+		return fmt.Errorf("failed to write controller deployment: %w", err)
+	}
+
+	dsTemplate, err := Sanitize(a.NodeTemplate)
+	if err != nil {
+		return fmt.Errorf("failed to sanitize node.yaml: %w", err)
+	}
+	if err := os.WriteFile(filepath.Join(path, NodeDaemonSetAssetName), dsTemplate, 0644); err != nil {
+		return fmt.Errorf("failed to write node daemonset: %w", err)
+	}
+
+	if err := saveAssets(path, a.ControllerStaticResources); err != nil {
+		return fmt.Errorf("failed to save controller static resources: %w", err)
+	}
+
+	if err := saveAssets(path, a.GuestStaticResources); err != nil {
+		return fmt.Errorf("failed to save guest static resources: %w", err)
+	}
+
+	if err := saveAssets(path, a.GuestStorageClassAssets); err != nil {
+		return fmt.Errorf("failed to save StorageClass assets: %w", err)
+	}
+
+	if err := saveAssets(path, a.GuestVolumeSnapshotClasses); err != nil {
+		return fmt.Errorf("failed to save VolumeSnapshotClass assets: %w", err)
+	}
+
+	return nil
+}
+
+func saveAssets(path string, assets map[string][]byte) error {
+	for name, assetBytes := range assets {
+		path := filepath.Join(path, name)
+		data, err := Sanitize(assetBytes)
+		if err != nil {
+			return fmt.Errorf("failed to sanitize asset %s: %w", name, err)
+		}
+		if err := os.WriteFile(path, data, 0644); err != nil {
+			return fmt.Errorf("failed to write asset %s: %w", name, err)
+		}
+	}
+	return nil
+}
+
+// NewFromAssets loads the assets from the given directory.
+func NewFromAssets(reader resourceapply.AssetFunc, dir string) (*CSIDriverAssets, error) {
```

Comment:
I couldn't figure our who and when calls `NewFromAssets()`. Ideas?
(it's worthy to have at least one caller, because strictly speaking its validity depends on `reader` arg -- hard to guess if it's simply `os.ReadFile()` or something more sophisticated)

Reply:
It will be used by a subsequent PR to actually load the files. See https://github.com/openshift/aws-ebs-csi-driver-operator/pull/258/files#diff-0d623dfd885adb20f991bda4c2453aebd732ca6dbb4d1d4be6e79805c3b48de6 and expand starter.go